lib/rucio/daemons/rsedecommissioner/profile.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Collection of functions that define the action of the decommissioning daemon on an RSE."""
+
+import logging
+from collections.abc import Callable, Iterable
+from dataclasses import dataclass
+from typing import Any, Union
+
+
+@dataclass
+class DecommissioningProfile:
```

Comment:
Because __init__.py imports functions from generic.py by name in the global scope, and the latter imports DecommissioningProfile, simply moving the class into __init__ would cause circular imports. So I made a few more changes and now __init__.py has a function that returns a profile instead of a dict of profiles in the global scope.

Reply:
Of course, you’re right. I missed this. Would `profiles/types.py` be better?

Reply:
I think that would improve the readability and avoids needing all the tricks. I added a `profiles/types.py` and reverted `__init__.py` to the original shape.